tableauserverclient/server/request_factory.py

```diff
@@ -30,12 +30,17 @@ def signin_req(self, auth_item):
 
 
 class DatasourceRequest(object):
-    def _generate_xml(self, datasource_item):
+    def _generate_xml(self, datasource_item, connection_credentials=None):
         xml_request = ET.Element('tsRequest')
         datasource_element = ET.SubElement(xml_request, 'datasource')
         datasource_element.attrib['name'] = datasource_item.name
         project_element = ET.SubElement(datasource_element, 'project')
         project_element.attrib['id'] = datasource_item.project_id
+        if connection_credentials:
```

Comment:
(not a shipblocker)
Since this is common code, we might consider pulling this into its own helper method.